common/include/common/state_counters.hpp

```diff
@@ -0,0 +1,267 @@
+/***************************************************************************
```

Comment:
Would it make sense to separate the states counters for different blas levels in different files? For example `blas1_state_counters.hpp`, `blas2_state_counters.hpp` and `blas3_state_counters.hpp`. We have this separation elsewhere in the codebase.